Reorder team seeding with drag/drop

SibiAkkash opened the pull request:
- [x] Drag drop to reorder team seed
- [x] Fix styles, show seed on the left, team Ids on the right.
Fixes #155 

https://github.com/india-ultimate/hub/assets/39093537/93754aac-c0ff-4f6f-a9e3-645e00fe97b2

![image](https://github.com/india-ultimate/hub/assets/39093537/f28a9a7f-1035-4151-bef3-fc3d738bf2be)

Joe2k commented:
Oof too good!!

punchagan commented:
This looks great! (from the video/screenshots). Thanks for working on this, @SibiAkkash 

What would it take to remove the Team IDs from the UI? Are the IDs only required in the pool creation input, or is there any other place too where they are useful? :thinking:

SibiAkkash commented:
> This looks great! (from the video/screenshots). Thanks for working on this, @SibiAkkash
> 
> What would it take to remove the Team IDs from the UI? Are the IDs only required in the pool creation input, or is there any other place too where they are useful? 🤔

Hi @punchagan it wouldn't take much to remove the Team IDs, i added them because i thought there could be some use for the IDs. For pool creation we only need the seed number, and not the team Ids. 
@Joe2k We can remove team IDs right ? I added them just in case

punchagan commented:
> Hi @punchagan it wouldn't take much to remove the Team IDs, i added them because i thought there could be some use for the IDs.

Yes, I understand that deleting the ID is just a matter of deleting a line of code. :smile: 

My question was whether you see the ID being used elsewhere (in the UI or in an offline workflow). If there's no other use of the IDs, I'd rather remove them to avoid confusion between ID numbers and seeds.

SibiAkkash commented:
> > Hi @punchagan it wouldn't take much to remove the Team IDs, i added them because i thought there could be some use for the IDs.
> 
> Yes, I understand that deleting the ID is just a matter of deleting a line of code. 😄
> 
> My question was whether you see the ID being used elsewhere (in the UI or in an offline workflow). If there's no other use of the IDs, I'd rather remove them to avoid confusion between ID numbers and seeds.

of course, why did i think you were asking about deleting a line of code 😅
afaik in the tournament manager, only seeds are used, and not team Ids

punchagan commented on frontend/src/components/TournamentManager.js:
```diff
@@ -39,13 +39,14 @@ import CreateTournamentForm from "./tournament/CreateTournamentForm";
   const [selectedTournament, setSelectedTournament] = createSignal();
   const [selectedTournamentID, setSelectedTournamentID] = createSignal(0);
-  const [tournamentSeeding, setTournamentSeeding] = createSignal([]);
+  const [teamSeeding, setTeamSeeding] = createSignal([]);
```
Is there a reason behind this name change? Just wanted to understand why you find this a better name.

SibiAkkash replied:
`tournamentSeeding` was being initialised to a dictionary previously. https://github.com/SibiAkkash/hub/blob/8f71f8edb2aba6e8af39d10883aa1fce7f20e9e8/frontend/src/components/TournamentManager.js#L86 and https://github.com/SibiAkkash/hub/blob/8f71f8edb2aba6e8af39d10883aa1fce7f20e9e8/frontend/src/components/TournamentManager.js#L370

punchagan replied:
Ah, makes sense! Just `teams` or `orderedTeams` might perhaps be better names?

SibiAkkash replied:
yes, i'll rename it to teams

punchagan commented on frontend/src/components/TournamentManager.js:
```diff
@@ -322,68 +325,48 @@ const TournamentManager = () => {
-                      <td class="px-6 py-4">{tournamentSeeding()[seed]}</td>
-                    </tr>
+            <div class="text-blue-500 text-xl font-bold mb-2">Seeding</div>
+            <div class="flex flex-row mb-5 w-1/2">
```
`w-1/2` should only be active on large screens. On mobile screens, the full width can be used.

SibiAkkash replied:
Ah yes, would some media queries solve this ?

punchagan replied:
You can add a prefix like `md` or `sm` to specify the size at which we want to start using a different rule. Something like would work, I think. 

```
<div class="w-1/2 md:w-full">
  <!-- ... -->
</div>
```

SibiAkkash replied:
yeah, i just added this. Thanks @punchagan

punchagan commented on frontend/src/components/tournament/ReorderTeams.js:
```diff
@@ -0,0 +1,92 @@
+        "transition-transform": !!state.active.draggable
+      }}
+    >
+      <TeamInfo teamId={props.teamId} teamName={props.teamsMap[props.teamId]} />
```
Having the team name + id rendered as separate elements, and the seeds as separate ones causes issues when the team name wraps around into two different lines. 

![image](https://github.com/india-ultimate/hub/assets/315678/b37f5276-418a-45cd-a411-a3dc6856770a)

SibiAkkash replied:
Yes, i did see this when i was messing around with the widths of the team name div.
- Removing the team name would give some space
- My thought process for rendering seeds separately was that, teams are being assigned seeds. Its really the teams moving, and not the seed values in itself.

I think a CSS fix is sufficient, i'll try to increase size of the team size div. what do you think ?

punchagan replied:
I agree with you on moving the teams around, but CSS fixes seem hacky to me; It may just be me. What if a team has a really long name - for instance `Learning to Fly` or `Breakside (Thatte Idli Kaal Soup)`. But, may be I'm overthinking this. Using the full width on smaller screens, and removing the team ID may be sufficient.

SibiAkkash replied:
yeah, i checked after removing teamID and full width on smaller screens. The issue might go away with this, although it is hacky. I've attached a video below, maybe you can check and see if that looks good ?

punchagan reviewed (changes requested):
Thanks for working on this, @SibiAkkash. It looks great! I have a couple of minor comments, and one slightly bigger issue. 

I think we can remove the team IDs from the UI and make them completely internal if they are not used anywhere. 

Rendering the seed as separate elements on the left seems to create issues when the row height of the seeds and the height of each row in the sortable list don't match. I wonder if the seed can also be a part of the team info, and dragEnd also updates the seeds displayed. Or may be we just improve the CSS to fix this.

Joe2k commented:
Yep team_id can be removed.. We only had it because of reordering the seeds via json. Now no longer needed

Joe2k commented on frontend/src/components/tournament/ReorderTeams.js:
```diff
@@ -0,0 +1,92 @@
+      const toIndex = currentItems.indexOf(droppable.id);
+      if (fromIndex !== toIndex) {
+        const updatedItems = currentItems.slice();
+        updatedItems.splice(toIndex, 0, ...updatedItems.splice(fromIndex, 1));
```
Here cant we run a loop through updatedItems and set teamSeed field in each element to the index of the element and display the teamSeed in the TeamInfo?

SibiAkkash commented:
@punchagan @Joe2k this is how it looks to drag/drop teams without a fixed number on the left. To me, it feels a bit weird, if you have a fixed number on the left, it feels like i'm drag/dropping a team to a "particular" seed. Idk how to explain it 😂

https://github.com/india-ultimate/hub/assets/39093537/bf5d77d7-2e0b-4953-affd-6365d79d4b31

Joe2k commented:
Haha this might be a bit weird when playing around but usualy comps already know how to seed the teams so this should be a fast solution for them

punchagan commented:
> Haha this might be a bit weird when playing around but usualy comps already know how to seed the teams so this should be a fast solution for them

The seeding also can have some styling like we have in the previous video/screenshots.

SibiAkkash commented:
> > Haha this might be a bit weird when playing around but usualy comps already know how to seed the teams so this should be a fast solution for them
> 
> 👍 The seeding also can have some styling like we have in the previous video/screenshots.

Yeah, i just wanted to show this recording to show y'all how it looks.
So i'll change it to this way now ?

SibiAkkash commented:
after changes
https://github.com/india-ultimate/hub/assets/39093537/a378dcce-408b-4747-90e0-b2f7245b6951

punchagan commented:
Thanks @SibiAkkash ! This is a great improvement for anyone creating tournaments. :clap:
